### PR TITLE
Added Michalewicz function

### PR DESCRIPTION
**Pull request by sebastianober**

### CODE REVIEW

File: tests/unit/objectives/test_single_objectives.py [hunk 1]
```diff
@@ -81,7 +92,7 @@ def test_objective_maps_minimizers_to_minimum(
 ) -> None:
     objective_values_at_minimizers = objective(minimizers)
     tf.debugging.assert_shapes([(objective_values_at_minimizers, [len(minimizers), 1])])
-    npt.assert_allclose(objective_values_at_minimizers, tf.squeeze(minimum), atol=1e-4)
+    npt.assert_allclose(objective_values_at_minimizers, tf.squeeze(minimum), atol=2e-4)
```
Comment (henrymoss) [inline tests/unit/objectives/test_single_objectives.py]: How many samples do you need to get within the existing tolerance?

File: tests/unit/objectives/test_single_objectives.py [hunk 2]
```diff
@@ -37,6 +37,13 @@
     LOGARITHMIC_GOLDSTEIN_PRICE_MINIMIZER,
     LOGARITHMIC_GOLDSTEIN_PRICE_MINIMUM,
     LOGARITHMIC_GOLDSTEIN_PRICE_SEARCH_SPACE,
+    MICHALEWICZ_2_MINIMIZER,
+    MICHALEWICZ_2_MINIMUM,
+    MICHALEWICZ_2_SEARCH_SPACE,
+    MICHALEWICZ_5_MINIMUM,
+    MICHALEWICZ_5_SEARCH_SPACE,
+    MICHALEWICZ_10_MINIMUM,
+    MICHALEWICZ_10_SEARCH_SPACE,
```
Comment (henrymoss) [inline tests/unit/objectives/test_single_objectives.py]: Where are the minimizers for 2 and 5?

Comment (sebastianober) [reply tests/unit/objectives/test_single_objectives.py]: I don't know them, they're not listed on the [https://www.sfu.ca/~ssurjano/michal.html](url) site...

File: tests/unit/objectives/test_single_objectives.py (hunk 1 above)
Comment (sebastianober) [reply tests/unit/objectives/test_single_objectives.py]: There aren't any samples here?

File: tests/unit/objectives/test_single_objectives.py (hunk 2 above)
Comment (henrymoss) [reply tests/unit/objectives/test_single_objectives.py]: Work them out? Probably useful to have them?

File: tests/unit/objectives/test_single_objectives.py (hunk 1 above)
Comment (henrymoss) [reply tests/unit/objectives/test_single_objectives.py]: My bad. So how come we are outside tolerance then???

**Review by henrymoss — APPROVED** [review]

File: tests/unit/objectives/test_single_objectives.py [hunk 3]
```diff
@@ -50,6 +59,9 @@
     hartmann_3,
     hartmann_6,
     logarithmic_goldstein_price,
+    michalewicz_2d,
```
Comment (hstojic) [inline tests/unit/objectives/test_single_objectives.py]: we should probably be consistent with naming of other functions and remove `d` part at the end in these 2d, 5d and 10d

**Review by hstojic — APPROVED** [review]
looks good, we would need to change the names slightly, see my comment above

File: tests/unit/objectives/test_single_objectives.py (hunk 3 above)
Comment (sebastianober) [reply tests/unit/objectives/test_single_objectives.py]: Good point, I've fixed this now. Anything else?

**Pull request merged**